### DISCUSSION TITLE
Implementing summation operator in PySR

### DISCUSSION THREAD

**Discussion by Arijit-Majumdar** [MilesCranmer/PySR#570] [Q&A]
Hi,
I am trying to use PySR to get expression for potential energy surface of molecules. My data is pairwise distance matrix and I want to perform summation operation along particular axis i.e. $\Sigma_j r_{ij}$ where $r_{ij}$ is my pairwise distance matrix. I am new to PySR and can't figure out how to implement $\Sigma_j$ operator. Any suggestions is greatly appreciated.

Thanks

**Comment by MilesCranmer** [comment]
Can you do this with a custom objective function that sums along an axis?

**Reply by Arijit-Majumdar** [reply]
I am not sure if it should be the objective function. My input data is a matrix and the output is a scalar. For example if my output is a function $f(r_{ij})$ and the true function is as follows
$f(r_{ij}) = \Sigma_i( (\Sigma_j r^6_{ij}) - (\Sigma_j {r^{12}}_{ij}))$

So I will need operators $\Sigma_i$, $\Sigma_j$ along with the +, - and ^ operators. Is it possible to build $\Sigma_i$ as a custom operator?

**Reply by MilesCranmer** [reply]
Ah, I see. Right now, this is not built-in. You would need to create a custom objective function that performs sums for specific operators.
